### PR TITLE
Feature/e UI 3670 new registration request happy unhappy path

### PR DESCRIPTION
**Pull request by tarikdogan-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3670

### Change description ###

New registration request-Select a radio button and clicks on Submit (Multiple PBA's)-Happy and Unhappy paths

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/org-manager/components/organisation-details-info/models/organisation-details.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  title: string;
+  description: string;
+  fieldId?: string;
+}
+export interface DisplayedRequest {
+  request: RequestType;
+  checked: boolean;
+}
+export enum RequestErrors {
+  NO_SELECTION = 'Select what would you like to do with this registration',
```
Comment (phillip-whitaker-hmcts) [inline src/org-manager/components/organisation-details-info/models/organisation-details.ts:16]: Where has this error text come from?

"Select what would you like.... "sounds a bit tongue-twisting, should it not be "Select what you would like..."

File: src/org-manager/components/organisation-details-info/organisation-details-info.component.ts [hunk 2]
```diff
@@ -8,14 +10,66 @@ import { OrganisationVM} from 'src/org-manager/models/organisation';
...
+          this.approveOrganisation(this.org);
+          break;
+        case RequestType.REJECT_REQUEST:
+          this.deleteOrganisation(this.org);
+          break;
+        default :
+          this.reviewOrganisation(this.org);
+          break;
+      }
+    }
```
Comment (phillip-whitaker-hmcts) [inline src/org-manager/components/organisation-details-info/organisation-details-info.component.ts:67]: Maybe to reduce nesting you could have...

```
if (this.formGroup.invalid) {
return;
}
const radioSelectedValue = this.formGroup.get(
  this.radioSelectedControlName
).value;

switch (radioSelectedValue) {
  case RequestType.APPROVE_REQUEST:
    this.approveOrganisation(this.org);
    break;
  case RequestType.REJECT_REQUEST:
    this.deleteOrganisation(this.org);
    break;
  default :
    this.reviewOrganisation(this.org);
    break;
```

File: src/org-manager/containers/review-organisation/review-organisation.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+  selector: 'app-org-pending-review',
+  templateUrl: './review-organisation.component.html'
+})
+export class ReviewOrganisationComponent implements OnInit {
+  public orgForReview: OrganisationVM | null;
+  public confirmButtonDisabled = false;
+
+  constructor(public store: Store<fromOrganisationPendingStore.OrganisationRootState>) {
```
Comment (phillip-whitaker-hmcts) [inline src/org-manager/containers/review-organisation/review-organisation.component.ts:17]: why are these dependencies public? Should they not be private readonly

File: src/org-manager/containers/review-organisation/review-organisation.component.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  public ngOnInit() {
```
Comment (phillip-whitaker-hmcts) [inline src/org-manager/containers/review-organisation/review-organisation.component.ts:20]: should always declare return type

File: src/org-manager/containers/review-organisation/review-organisation.component.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    this.addOrganisationForReviewSubscribe();
+  }
+
+  /**
+   * Add Organisation For Review Subscribe
+   *
+   * We subscribe to the organisation under review, so that we can display this information to the user within the view.
+   */
+  public addOrganisationForReviewSubscribe() {
```
Comment (phillip-whitaker-hmcts) [inline src/org-manager/containers/review-organisation/review-organisation.component.ts:29]: return type

File: src/org-manager/store/effects/organisation.effects.ts [hunk 6]
```diff
@@ -194,6 +194,15 @@ export class OrganisationEffects {
     })
   );
 
+
+  @Effect()
+  public navToReviewOrganisation$ = this.actions$.pipe(
+    ofType(pendingOrgActions.OrgActionTypes.NAV_TO_REVIEW_ORGANISATION),
+    map(() => {
+      return new fromRoot.Go({ path: ['/review-organisation'] });
+    })
```
Comment (phillip-whitaker-hmcts) [inline src/org-manager/store/effects/organisation.effects.ts:203]: semicolon;

**Review by phillip-whitaker-hmcts — CHANGES_REQUESTED** [review]

File: src/org-manager/components/organisation-details-info/models/organisation-details.ts (hunk 1 above)
Comment (tarikdogan-hmcts) [reply src/org-manager/components/organisation-details-info/models/organisation-details.ts:16]: It comes from the requirements from the ticket (scenario :4)

File: src/org-manager/containers/review-organisation/review-organisation.component.ts (hunk 5 above)
Comment (tarikdogan-hmcts) [reply src/org-manager/containers/review-organisation/review-organisation.component.ts]: done

File: src/org-manager/store/effects/organisation.effects.ts (hunk 6 above)
Comment (tarikdogan-hmcts) [reply src/org-manager/store/effects/organisation.effects.ts:203]: same implementation with other effects, so to be consisted in remaining part of the code it left.

File: src/org-manager/components/organisation-details-info/organisation-details-info.component.ts (hunk 2 above)
Comment (tarikdogan-hmcts) [reply src/org-manager/components/organisation-details-info/organisation-details-info.component.ts:68]: done

File: src/org-manager/containers/organisation-details/organisation-details.component.ts [hunk 7]
```diff
@@ -99,6 +99,12 @@ export class OrganisationDetailsComponent implements OnInit, OnDestroy {
     }
   }
 
+  public reviewOrganisation(data: OrganisationVM) {
```
Comment (kuda1992) [inline src/org-manager/containers/organisation-details/organisation-details.component.ts:102]: Missing return type

File: src/org-manager/containers/review-organisation/review-organisation.component.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import {OrganisationVM} from '../../../org-manager/models/organisation';
+import * as fromOrganisationPendingStore from '../../../org-manager/store';
+import {getOrganisationForReview} from '../../store/selectors';
+
+@Component({
+  selector: 'app-org-pending-review',
+  templateUrl: './review-organisation.component.html'
+})
+export class ReviewOrganisationComponent implements OnInit {
+  public orgForReview: OrganisationVM | null;
```
Comment (kuda1992) [inline src/org-manager/containers/review-organisation/review-organisation.component.ts:14]: Why is the other type null?

File: src/org-manager/containers/review-organisation/review-organisation.component.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    this.addOrganisationForReviewSubscribe();
+  }
+
+  /**
+   * Add Organisation For Review Subscribe
+   *
+   * We subscribe to the organisation under review, so that we can display this information to the user within the view.
+   */
+  public addOrganisationForReviewSubscribe(): void {
+    this.store.pipe(select(getOrganisationForReview), take(1)).subscribe((org: OrganisationVM) => {
```
Comment (kuda1992) [inline src/org-manager/containers/review-organisation/review-organisation.component.ts:30]: Why are we not subscribing to this selector?

**Review by kuda1992 — CHANGES_REQUESTED** [review]
Small minor comments to addess

**Review by kuda1992 — APPROVED** [review]
Looks good

**Pull request merged**